sonar-plugin-api/src/main/java/org/sonar/api/platform/Server.java

```diff
@@ -27,24 +27,62 @@
 import org.sonar.api.server.ServerSide;
 
 /**
+ * Runtime information about server
+ *
  * @since 2.2
-  */
+ */
 @ScannerSide
 @ServerSide
 @ComputeEngineSide
 public abstract class Server {
 
+  /**
+   * Name is misleading, this is an UUID generated
+   * at each startup, so it changes if server is restarted.
+   * @return a non-null UUID. Format can change over versions.
+   */
   public abstract String getId();
 
+  /**
+   * UUID generated on demand by system administrators. It is
+   * {@code null} by default on fresh installations. When defined,
+   * value does not change when server is restarted.
+   * @since 2.10
+   */
+  @CheckForNull
+  public abstract String getPermanentServerId();
+
+  /**
+   * Non-null version of SonarQube at runtime
+   */
   public abstract String getVersion();
 
+  /**
+   * Date when server started
+   */
   public abstract Date getStartedAt();
 
+  /**
+   * @deprecated in 6.0. Replaced by {@link ServerFileSystem#getHomeDir()}
+   * @return an existing directory in server and CE environments, {@code null} in scanner.
+   */
+  @Deprecated
   public abstract File getRootDir();
 
+  /**
+   * @deprecated always {@code null} since version 6.0. No alternatives, as plugins do not have to touch this directory.
+   */
+  @Deprecated
   @CheckForNull
   public abstract File getDeployDir();
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)